### PR TITLE
WIP: iOS performance speedup with Bender and MPS

### PR DESCRIPTION
**Pull request by mats-claassen**
Please review:
* Should I mention DreamSnap or not?
* Conclusions and CTA
* Title suggestions welcome
* Should I go into more detail at some point
* Presentation of benchmark results

### CODE REVIEW

**Review by bryant1410 — COMMENTED** [review]
I didn't read it, just looked at it superficially. I think it'd be good to show the benchmark results on a bar/line plot. It's visually clearer and easier to understand the difference.

**Comment by bryant1410** [comment]
Also, what if converting the section "So which framework should I choose?" into a list of bullet points of "when to choose bender" and "when to choose Core ML"? So it's like a reference that people come back to read again at other times. Having it as bullet points helps with this goal.

**Comment by mtnbarreto** [comment]
> I didn't read it, just looked at it superficially. I think it'd be good to show the benchmark results on a bar/line plot. It's visually clearer and easier to understand the difference.

agreed! we can manage this with the design team (pesto).

File: _posts/2021-08-30-bender-is-the-best.md [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+The network architecture will likely be an important factor here.
+For certain architectures the speedup will be more while for others you might not get any benefit using MPS.
+
+## To sum up
+
+CoreML and TFlite are easy to use frameworks but may not be the best options if you need realtime video processing.
+This is where MPS or Bender come in to help with increased performance, albeit with greater battery use.
+
+Bender allows converting frozen TF models and running them in iOS, although, this process might not be straightforward.
+If you need help doing such a conversion feel free to reach out at [hi@xmartlabs.com](mailto:hi@xmartlabs.com)
```
Comment (renzodgc) [inline _posts/2021-08-30-bender-is-the-best.md:109]:
```suggestion
That being said we created Bender back then to have a friendlier interface to MPS, if you need help doing such a conversion feel free to reach out at [hi@xmartlabs.com](mailto:hi@xmartlabs.com).
```

File: _posts/2021-08-30-bender-is-the-best.md [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+Have you ever wanted to run a neural network model on iOS and found out that you can't get it to run fast enough?
+We have faced this problem and gone through several options to make models run in a performant way. We found out that CoreML or TFlite are not always the best options but that in some cases using MetalPerformanceShaders (MPS) or Bender is the way to go.
+
+In this post we will go through our journey on deciding which framework to use, show performance comparisons and hopefully help you decide which one to use.
+
+> Note: In this post we didn't consider MLKit but it's performance should be similar to using TFlite directly.
+
+## Running style transfer on realtime video in iOS
+A few years ago, before the release of CoreML and TFlite on iOS, we built [DreamSnap], an app that runs style transfer on camera input in realtime and let's users take stylized photos or videos.
+Now we decided we wanted to update the app with newer models and found a Magenta model hosted on [TFHub].
```
Comment (pgrill) [inline _posts/2021-08-30-bender-is-the-best.md:23]: Maybe we can include a short description of what Magenta is?

**Review by mtnbarreto — CHANGES_REQUESTED** [review]

**Pull request merged**